src/main/resources/ui/controller/UpsertWorkflow.controller.js

```diff
@@ -104,14 +104,27 @@ sap.ui.define([
 		},
 
 		onSaveWorkflow: function () {
-			let isEdit = this._model.getProperty("/isEdit");
-			let workflow = this.getWorkflowToSave();
-			if(isEdit) {
-				WorkflowRepository.updateWorkflow(workflow);
-			} else {
-				WorkflowRepository.createWorkflow(workflow);
+			if(this._isWorkflowValid(this._model.getProperty("/workflow"))){
+				let isEdit = this._model.getProperty("/isEdit");
+				let workflow = this.getWorkflowToSave();
+				if(isEdit) {
+					WorkflowRepository.updateWorkflow(workflow);
+				} else {
+					WorkflowRepository.createWorkflow(workflow);
+				}
+				this.myNavBack();
 			}
-			this.myNavBack();
+		},
+
+		_isWorkflowValid: function (workflow) {
+			try {
+				cronstrue.toString(workflow.sensor.properties.settings.variables.cronExpression);
```

Comment:
You could add a custom validation here for that specific case that cronstrue checks wrongly (one of day of month and day of week must be `?`)

Reply:
In fact, please do

Reply:
If you do it as in Enceladus and let user only select what is possible, then you can skip the validations. 

BTW Why don't you use the output of the `toString` function to show to the users, so they know what time they have set up. This is one of the really convenient features for the user

Reply:
Kevin - yes, fixed
Sasa - 1. Yes in Angular UI. 2. Yes, we do.